tesla-hero-slider/src/components/slide.js

```diff
@@ -0,0 +1,71 @@
+import React, {Component} from 'react';
+import propTypes from 'prop-types';
+import {TransitionGroup, CSSTransition} from 'react-transition-group';
+import SlideParams from './slide-params';
+import SlideAside from './slide-aside';
+import './../styles/scss/components/slide.css';
+
+
+class Slide extends Component {
+
+	static propTypes = {
+		activeSlide: propTypes.object.isRequired,
+		animationForward: propTypes.bool.isRequired,
+		setAnimationState: propTypes.func.isRequired,
+		ANIMATION_PHASES: propTypes.object.isRequired
+	};
+
+	handleTransitionEnd(e) {
+		//checking both elements if they do not have classes -enter/-exit
+		if (e.currentTarget.className === 'tesla-slide__img'
```

Comment:
You can use CSSTransition lifecycle methods to determinate a state of an animation: https://reactcommunity.org/react-transition-group/#CSSTransition-prop-onExit